### ISSUE TITLE
array-type rule requires array in places that are less readable

### ISSUE THREAD

**Issue by alexeagle** [palantir/tslint#1526]
### Bug Report

- __TSLint version__: HEAD

#### TypeScript code being linted

```ts
Array<{length: number, deferred: Deferred<any[]>}>
Array<Document|string>
Array<(event: Event) => void>
TableRow<Array<string|number|null>>
```

with `tslint.json` configuration:

```json
(include if relevant)
{
  "rules": {
    "array-type": [true, "array"]
  }
}
```

#### Actual behavior

Replaces with
```ts
{length: number, deferred: Deferred<any[]>}[]
(Document|string)[]
((event: Event) => void)[]
TableRow<(string|number|null)[]>
```

#### Expected behavior

These are more readable in the original form with `Array<T>`. 

@mprobst explains:
precisely, if your type in the grammar (https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#a1-types) is not just a PredefinedType (number, string, etc) or a TypeReference (some.namepsace.MyFunkyType), then you should be allowed to use Array<MyType>.

**Comment by JKillian** [comment]
Sounds like you'd like to follow the same guidelines that @danvk mentioned [here](https://github.com/palantir/tslint/pull/1498#issuecomment-241144177):

> Not necessarily something that needs to be implemented in this PR, but our style guide is a little more nuanced than either of these options. For simple (non-generic) types we use `T[]`. If the type becomes more complex, we switch to `Array<>` for clarity, e.g. `Array<{[key: string]: T}>`.

**Comment by alexeagle** [comment]
Yes, exactly, I read that comment on the original issue but didn't think deeply enough. @ScottSWu do you want to do the `"array-type": [true, "array-for-simple-types"]` or shall I?
